### PR TITLE
add pull request template

### PR DESCRIPTION
**Pull request by tenmozes**
Add Pull request template with the following description

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### Checklist (Optional for External Contributions)

- [ ] Include a link to the GitHub issue in the commit message, if issue exists.
- [ ] Mention the change in the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md). Explain what has changed and why. If there is a related issue or documentation change - link them as well.

  Tips for writing a good changelog message::

    * Write a human-readable changelog message that describes the problem and solution.
    * Include a link to the issue or pull request in your changelog message.
    * Use specific language identifying the fix, such as an error message, metric name, or flag name.
    * Provide a link to the relevant documentation for any new features you add or modify.

- [ ] After your pull request is merged, please add a message to the issue with instructions for how to test the fix or try the feature you added. Here is an [example](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4048#issuecomment-1546453726)
- [ ] Do not close the original issue before the change is released. Please note, in some cases Github can automatically close the issue once PR is merged. Re-open the issue in such case.
- [ ] If the change somehow affects public interfaces (a new flag was added or updated, or some behavior has changed) - add the corresponding change to documentation.

Examples of good changelog messages:

1. FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent.html): add support for [VictoriaMetrics remote write protocol](https://docs.victoriametrics.com/vmagent.html#victoriametrics-remote-write-protocol) when [sending / receiving data to / from Kafka](https://docs.victoriametrics.com/vmagent.html#kafka-integration). This protocol allows saving egress network bandwidth costs when sending data from `vmagent` to `Kafka` located in another datacenter or availability zone. See [this feature request](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1225).

2. BUGFI

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+### Describe Your Changes
+
+Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.
+
+### Checklist (Optional for External Contributions)
+
+- [ ] Include a link to the GitHub issue in the PR description, if possible.
```
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:7]: PR description or commit message? I think it should be the latter.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Add an item to the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md) that describes the problem you're addressing and how you're solving it.
```
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:8]:
```suggestion
- [ ] Mention the change in the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md). Explain what has changed and why. If there is a related issue or documentation change - link them as well.
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  Tips for writing a good changelog message::
+
+    * Write a human-readable changelog message that describes the problem and solution.
+    * Include a link to the issue or pull request in your changelog message.
+    * Use specific language identifying the fix, such as an error message, metric name, or flag name.
+    * Provide a link to the relevant documentation for any new features you add or modify.
```
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:15]:
```suggestion
```

I don't think someone needs to read it on each PR.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:8]: Please add the following:

```
- [ ] If the change somehow affects public interfaces (a new flag was added or updated, or some behavior has changed) - add the corresponding change to documentation.
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+- [ ] After your pull request is merged, please add a message to the issue with instructions for how to test the fix or try the feature you added.
```
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: WOuld you mind adding an example of such message?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Please avoid closing the original issue before the release unless you are the author of the issue and want to close it or the fix/feature doesn't require the release procedure.
```
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:18]:
```suggestion
- [ ] Do not close the original issue before the change is released. Please note, in some cases Github can automatically close the issue once PR is merged. Re-open the issue in such case.
```

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 3 above)
Comment (tenmozes) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:15]: this message has several goals 
- provide information for every new engineer
- remind everyone who create PRs to do some steps

so i don't think this repeatable PR to PR information is useless

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] Mention the change in the [Changelog](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/CHANGELOG.md). Explain what has changed and why. If there is a related issue or documentation change - link them as well.
+
+  Tips for writing a good changelog message::
+
+    * Write a human-readable changelog message that describes the problem and solution.
+    * Include a link to the issue or pull request in your changelog message.
+    * Use specific language identifying the fix, such as an error message, metric name, or flag name.
+    * Provide a link to the relevant documentation for any new features you add or modify.
+
+- [ ] After your pull request is merged, please add a message to the issue with instructions for how to test the fix or try the feature you added. Here is an [example](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4048#issuecomment-1546453726)
```
Comment (hagen1778) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]:
```suggestion
- [ ] After your pull request is merged, please add a message to the original issue with instructions on how to test the change. See the message example [here](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4048#issuecomment-1546453726).
```

Comment (hagen1778) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: bump

Comment (hagen1778) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: bump

**Comment by hagen1778** [comment]
@tenmozes should we close this pr?

**Comment by tenmozes** [comment]
@hagen1778 nope, will take a look soon

**Comment by hagen1778** [comment]
@tenmozes @zekker6 should we close it in favour of https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4315/commits/ecc970addcc1d648f1a04024858f6b7e4c89cd8e ?

**Comment by zekker6** [comment]
@hagen1778 Yes, looks good to me. See also:
https://github.com/VictoriaMetrics/VictoriaMetrics/commit/c52f73b3e19500946bb22c7cf620f82b24d591f8
https://github.com/VictoriaMetrics/VictoriaMetrics/commit/bbdd7743a8265087d0523a95f8ff4285b1f356b4
https://github.com/VictoriaMetrics/VictoriaMetrics/commit/0e75581dd96f798784d6b19525846c752daf804a

**Comment by hagen1778** [comment]
@tenmozes I'm closing this PR. Feel free to send the new PR if you think something is missing in the commits mentioned above.

**Pull request closed without merging**